libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -206,16 +204,17 @@ private void initNavigationLocationEngine() {
     locationEngineConductor = new LocationEngineConductor(locationEngineCallback);
   }
 
-  private void initLanguage(NavigationUiOptions options, LocaleUtils localeUtils) {
-    language = localeUtils.getNonEmptyLanguage(getApplication(), options.directionsRoute().voiceLanguage());
+  private void initLanguage(NavigationUiOptions options) {
+    language = options.directionsRoute().routeOptions().language();
   }
 
-  private void initUnitType(NavigationUiOptions options, LocaleUtils localeUtils) {
+  private void initUnitType(NavigationUiOptions options) {
     unitType = options.directionsRoute().routeOptions().voiceUnits();
   }
 
-  private void initVoiceInstructions() {
-    instructionPlayer = new NavigationInstructionPlayer(getApplication(), language, accessToken);
+  private void initVoiceInstructions(NavigationViewOptions options) {
+    boolean languageSupported = options.directionsRoute().voiceLanguage() != null;
```

Comment:
I guess this check should be included within `NavigationInstructionPlayer`. This way we remove this local variable and a parameter from `NavigationInstructionPlayer`. Does that make sense?

Reply:
@Guardiola31337 You would still need the options or the directionsRoute in the `NavigationInstructionPlayer` to make that check, which I don't think makes sense

Reply:
Sorry, I thought that we were using `language`, then I saw that we have `options.directionsRoute().voiceLanguage()` and `options.directionsRoute().routeOptions().language()`.